### PR TITLE
Bug/pdct 1143 update event list

### PR DESCRIPTION
**Pull request by annaCPR**
# What's changed
- added a fake API and repository for mocking calls in tests
- extracted family routes to a separate file so they can be tested
- added a render function for routes so we can test them easily
- added a test to reproduce the bug
- fixed the bug by reloading the UseEvent hook after edit

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

File: src/hooks/useEvent.ts [hunk 1]
```diff
@@ -8,29 +8,38 @@ const useEvent = (id?: string) => {
...
-        .catch((error: IError) => {
-          setError(error)
-        })
-        .finally(() => {
-          setLoading(false)
-        })
-    }
+  const reload = useCallback(() => {
+    handleGetEvent(false)
+  }, [handleGetEvent])
```
Comment (PatFawbertMills) [inline src/hooks/useEvent.ts:32]: Not sure using a useCallback function as a dependancy for the useCallback is a good standard, can we use the `id` same as the useCallback above?

File: src/routes/index.tsx [hunk 2]
```diff
@@ -37,29 +33,7 @@ const authenticatedRoutes = [
...
-                children: [
-                  {
-                    path: '',
-                    element: <FamilyList />,
-                    loader: familiesLoader,
-                    errorElement: <ErrorPage />,
-                  },
-                ],
-              },
+              ...familyRoutes,
```
Comment (PatFawbertMills) [inline src/routes/index.tsx:36]: Nice

File: src/tests/mocks/handlers.ts [hunk 3]
```diff
@@ -1,9 +1,49 @@
-import { http } from 'msw'
+import { http, HttpResponse } from 'msw'
```
Comment (PatFawbertMills) [inline src/tests/mocks/handlers.ts:1]: Look forward to the demo!

**Review by PatFawbertMills — CHANGES_REQUESTED** [review]

File: src/hooks/useEvent.ts (hunk 1 above)
Comment (annaCPR) [reply src/hooks/useEvent.ts:32]: Good catch! I only added handleGetEvent as a dependency cos the linter complains. Any ideas how else I can get around it?

Comment (PatFawbertMills) [reply src/hooks/useEvent.ts:32]: Actually having read up I think this is fine!

**Review by PatFawbertMills — APPROVED** [review]
Nice updates! 🔨

**Pull request merged**